pkg/crc/cluster/cluster.go

```diff
@@ -198,3 +198,44 @@ func addPullSecretToInstanceDisk(sshRunner *ssh.SSHRunner, pullSec string) error
 	}
 	return nil
 }
+
+func WaitforRequestHeaderClientCaFile(oc oc.OcConfig) error {
+	cmdArgs := []string{"get", "configmaps/extension-apiserver-authentication", `-ojsonpath={.data.requestheader-client-ca-file}`,
+		"-n", "kube-system"}
+
+	if err := oc.WaitForOpenshiftResource("configmaps"); err != nil {
+		return err
+	}
+
+	// Timeout after 180 seconds
+	for i := 0; i < 90; i++ {
+		stdout, stderr, err := oc.RunOcCommand(cmdArgs...)
+		if err != nil {
+			return fmt.Errorf("Failed to get request header client ca file %v: %s", err, stderr)
+		}
+		logging.Debug(stdout)
+		if stdout != "" {
+			break
+		}
+		time.Sleep(2 * time.Second)
```

Comment:
Could this use `errors.RetryAfter`?

Reply:
`errors.RetryAfter` is used in case of error retry but here we are retry because of the stdout shouldn't be empty string, error can be nil in this case since the configmap resource is available and command will successful by returning an empty string.

Reply:
but nothing prevents us from returning our own RetriableError when stdout is empty?
untested, but something like this:
```
diff --git a/pkg/crc/cluster/cluster.go b/pkg/crc/cluster/cluster.go
index 73a755a6..cabb9238 100644
--- a/pkg/crc/cluster/cluster.go
+++ b/pkg/crc/cluster/cluster.go
@@ -198,3 +198,44 @@ func addPullSecretToInstanceDisk(sshRunner *ssh.SSHRunner, pullSec string) error
 	}
 	return nil
 }
+
+func WaitforRequestHeaderClientCaFile(oc oc.OcConfig) error {
+	if err := oc.WaitForOpenshiftResource("configmaps"); err != nil {
+		return err
+	}
+
+	lookupRequestHeaderClientCa := func() error {
+		cmdArgs := []string{"get", "configmaps/extension-apiserver-authentication", `-ojsonpath={.data.requestheader-client-ca-file}`,
+			"-n", "kube-system"}
+
+		stdout, stderr, err := oc.RunOcCommand(cmdArgs...)
+		if err != nil {
+			return fmt.Errorf("Failed to get request header client ca file %v: %s", err, stderr)
+		}
+		if stdout == "" {
+			return &errors.RetriableError{Err: fmt.Errorf("missing .data.requestheader-client-ca-file")}
+		}
+		logging.Debugf("Found .data.requestheader-client-ca-file: %s", stdout)
+		return nil
+	}
+
+	return errors.RetryAfter(90, lookupRequestHeaderClientCa, 2*time.Second)
+}
```

Reply:
@cfergeau yes if we create our own error message then we can use it, as you described. I don't want to use it this non error message but if you insist I can just add same :)

Reply:
Not going to insist, was just a suggestion ;) If you prefer it without RetryAfter, fine with me.